src/mumble/AudioOutput.cpp

```diff
@@ -531,32 +542,36 @@ bool AudioOutput::mix(void *outbuff, unsigned int nsamp) {
 				}
 
 				if (prioritySpeakerActive) {
-					
+
 					if (user->tsState != Settings::Whispering
 					    && !user->bPrioritySpeaker) {
-						
+
 						volumeAdjustment *= adjustFactor;
 					}
 				}
 			}
 
 			// If recording is enabled add the current audio source to the recording buffer
 			if (recorder) {
-				AudioOutputSpeech *aos = qobject_cast<AudioOutputSpeech *>(aop);
-
-				if (aos) {
-					for (unsigned int i = 0; i < nsamp; ++i) {
-						recbuff[i] += pfBuffer[i] * volumeAdjustment;
+				if (speech) {
+					if (speech->bStereo) {
+						for (unsigned int i = 0; i < frameNumber; ++i) { // Mix down stereo to mono. TODO: stereo record support
+							recbuff[i] += (pfBuffer[2*i] / 2.0 + pfBuffer[2*i+1] / 2.0) * volumeAdjustment;
```

Comment:
A comment about a frame in this case being `[LR]` might be helpful here for future contributors